debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorTask.java

```diff
@@ -97,7 +91,26 @@ public String version() {
 
         this.schema = new MySqlDatabaseSchema(connectorConfig, valueConverters, topicSelector, schemaNameAdjuster, tableIdCaseInsensitive);
 
+        LOGGER.info("Closing connection before starting schema recovery");
+
+        try {
+            connection.close();
+        }
+        catch (SQLException e) {
+            throw new DebeziumException(e);
+        }
+
         validateAndLoadDatabaseHistory(connectorConfig, previousOffset, schema);
+
+        LOGGER.info("Reconnecting after finishing schema recovery");
+
+        try {
+            connection.setAutoCommit(false);
```

Comment:
Don't we have to re-create the connection here? I.e. isn't a connection dis-functional once being closed?

Reply:
It will automatically reconnect in https://github.com/debezium/debezium/blob/ba581ed323326618d32e8d4509d72c1e5d813636/debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java#L875-L877

Reply:
Ah, yeah, forgot there'd be that re-connect functionality in our wrapper :)